### PR TITLE
Updates HA kubeadm docs

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

### CODE REVIEW

File: content/en/docs/setup/independent/high-availability.md [hunk 1]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
+## Planning
 
-    {{< note >}}
-    **Optional:** You can modify `ca-csr.json` to add a section for `names`.
-    See [the CFSSL wiki](https://github.com/cloudflare/cfssl/wiki/Creating-a-new-CSR) for an example.
-    {{< /note >}}
+1. Figure out your pod CIDR. Please read the [CNI network
+   section](/docs/setup/independent/create-cluster-kubeadm/#pod-network) of the
+   docs for more information. Some CNI providers do not require a value to be
+   set. This guide uses calico for no particular reason. Because of this choice
```
Comment (rdodev) [inline content/en/docs/setup/independent/high-availability.md]: I would phrase this in a more positive way: 

"Some CNI providers do not require a value to be  set. This guide uses calico for no particular reason."

To something like:

 In this in example we have chose to use Calico (please note that some cni providers do not require special pod CIDR).

Comment (mistyhacks) [reply content/en/docs/setup/independent/high-availability.md]: Feedback on the feedback, I would not state that you're using Calico for "no particular reason."

File: content/en/docs/setup/independent/high-availability.md [hunk 2]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
+After this step, you have a resolvable DNS name for your load balancer.
+
+You should add the control plane nodes to the load balancer but be aware they
+will fail the healthcheck until the kube-apiserver is running.
+
+### The first control plane node
+
+#### Prerequisites
+
+1. This node must have [kubeadm, kubelet and kubectl][installation] installed.
```
Comment (rdodev) [inline content/en/docs/setup/independent/high-availability.md]: I'd link and list individually:

This node must have installed 
 
* [kubeadm]()
* [kubelet]()
* [kubectl]()

File: content/en/docs/setup/independent/high-availability.md [hunk 3]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
+```
 
-    - `<private-ip>` with the private IPv4 of the master server.
-    - `<etcd0-ip>`, `<etcd1-ip>` and `<etcd2-ip>` with the IP addresses of your three etcd nodes
-    - `<podCIDR>` with your Pod CIDR. Please read the [CNI network section](/docs/setup/independent/create-cluster-kubeadm/#pod-network) of the docs for more information. Some CNI providers do not require a value to be set.
-    - `<load-balancer-ip>` with the virtual IP set up in the load balancer. Please read [setting up a master load balancer](/docs/setup/independent/high-availability/#set-up-master-load-balancer) section of the docs for more information.
+Everything is now in place so you can run `kubeadm init --config kubeadm-config.yaml`
 
-    **Note:** If you are using Kubernetes 1.9+, you can replace the `apiserver-count: 3` extra argument with `endpoint-reconciler-type: lease`. For more information, see [the documentation](/docs/admin/high-availability/#endpoint-reconciler).
+### Swizzling certs
```
Comment (rdodev) [inline content/en/docs/setup/independent/high-availability.md]: I'd add some flavor text along the lines of:

"For those unfamiliar with the term "swizzling" in this context, it essentially refers to copying the certificates generated  in the steps above to other master hosts."

File: content/en/docs/setup/independent/high-availability.md (hunk 1 above)
Comment (rdodev) [reply content/en/docs/setup/independent/high-availability.md]: @mistyhacks exactly. Which is I why I'm suggesting he rephrased it to something more affirmative.

Comment (rdodev) [reply content/en/docs/setup/independent/high-availability.md]: I know we don't want to "pick winners" or "take sides", but there are very valid reason why Calico is a great choice for most use cases.

**Comment by Bradamant3** [comment]
/assign

(kubeadm-related)

File: content/en/docs/setup/independent/high-availability.md (hunk 3 above)
Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: Better not to use the term at all. Too much insider-speak. A better option (in use elsewhere in the docs, too): "Copy certificates" (don't need gerund form of verb, either). In answer to Ruben's question on Slack: "certificates" is preferable because clearer to non-native speakers/readers of English; abbreviations are not so reader-friendly in general. Yes, "certs" appears all over the docs, but we fix what we can when we encounter it.

File: content/en/docs/setup/independent/high-availability.md (hunk 1 above)
Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: The whole paragraph/step doesn't hold together very well, and any explanations are extraneous to a task-baseed doc. Try something like:
"1.  Find your pod CIDR. For details, see [the CNI network documentation](/docs/setup/independent/create-cluster-kubeadm/#pod-network). This example uses Calico, so the pod CIDR is `192.168.0.0/16`."

File: content/en/docs/setup/independent/high-availability.md [hunk 4]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-     ```bash
-     systemctl daemon-reload
-     systemctl start etcd
-     ```
-
-1. Check that it launched successfully:
-     ```bash
-     systemctl status etcd
-     ```
+Now we can take that value and put it in `~/.ssh/authorized_keys` of any other host we want to connect to.
```
Comment (timothysc) [inline content/en/docs/setup/independent/high-availability.md]: I'd add a snippet folks can use to copy keys around.  

```
for host in m1.example.com \
    m2.example.com \
    m3.example.com; \
    do ssh-copy-id -i ~/.ssh/id_rsa.pub $host; \
    done
```

File: content/en/docs/setup/independent/high-availability.md [hunk 5]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-
-    Replace the `<VIRTUAL-IP>` by your chosen virtual IP.
+{{% capture steps %}}
 
-4. Restart keepalived. While no Kubernetes services are up yet it will log health check fails on all master nodes. This will stop as soon as the first master node has been bootstrapped.
-{{% /tab %}}
-{{< /tabs >}}
+# External Etcd -- The instructions
 
+This guide assumes calico will be used as the pod network and will therefore use
```
Comment (timothysc) [inline content/en/docs/setup/independent/high-availability.md]: Why is this here?

File: content/en/docs/setup/independent/high-availability.md [hunk 6]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-     networking:
-      podSubnet: <podCIDR>
-     apiServerCertSANs:
-     - <load-balancer-ip>
-     apiServerExtraArgs:
-      apiserver-count: "3"
-     EOF
-     ```
+```
+scp /etc/kubernetes/pki/etcd/ca.crt ubuntu@ETCD_HOST_1:/home/ubuntu
```
Comment (timothysc) [inline content/en/docs/setup/independent/high-availability.md]: nix the ubuntu name imo , and keep is generic `user`

File: content/en/docs/setup/independent/high-availability.md [hunk 7]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
+```
 
-    - `<private-ip>` with the private IPv4 of the master server.
-    - `<etcd0-ip>`, `<etcd1-ip>` and `<etcd2-ip>` with the IP addresses of your three etcd nodes
-    - `<podCIDR>` with your Pod CIDR. Please read the [CNI network section](/docs/setup/independent/create-cluster-kubeadm/#pod-network) of the docs for more information. Some CNI providers do not require a value to be set.
-    - `<load-balancer-ip>` with the virtual IP set up in the load balancer. Please read [setting up a master load balancer](/docs/setup/independent/high-availability/#set-up-master-load-balancer) section of the docs for more information.
+Everything is now in place so you can run `kubeadm init --config kubeadm-config.yaml`
 
-    **Note:** If you are using Kubernetes 1.9+, you can replace the `apiserver-count: 3` extra argument with `endpoint-reconciler-type: lease`. For more information, see [the documentation](/docs/admin/high-availability/#endpoint-reconciler).
+### Swizzling certs
```
Comment (timothysc) [inline content/en/docs/setup/independent/high-availability.md]: Let's just call is copying, that term is non-standard.

Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: YES THANK YOU TIM

File: content/en/docs/setup/independent/high-availability.md (hunk 4 above)
Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: Yes. This should be another numbered step, too:

1. Add the returned value to `~/.ssh/authorized_keys` on the other hosts to connect to:
<Tim's snippet>

File: content/en/docs/setup/independent/high-availability.md [hunk 8]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-
-3. Install the following health check script to _/etc/keepalived/check_apiserver.sh_ on all master nodes:
-     ```bash
-     #!/bin/sh
-
-     errorExit() {
-        echo "*** $*" 1>&2
-        exit 1
-     }
+At the end of this step, try `ssh`ing between nodes to ensure success.
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: STEP! (are you getting tired of this yet? ;)):

1. To test, SSH between nodes.

File: content/en/docs/setup/independent/high-availability.md [hunk 9]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-     fi
-     ```
-
-    Replace the `<VIRTUAL-IP>` by your chosen virtual IP.
+{{% capture steps %}}
 
-4. Restart keepalived. While no Kubernetes services are up yet it will log health check fails on all master nodes. This will stop as soon as the first master node has been bootstrapped.
-{{% /tab %}}
-{{< /tabs >}}
+# External Etcd -- The instructions
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: h2, plus appropriately tagged children. Also try "Set up external etcd"

File: content/en/docs/setup/independent/high-availability.md [hunk 10]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
 
+This guide assumes calico will be used as the pod network and will therefore use
+some default values, but they will be called out explicitly so replacing calico
+with another pod network is possible.
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: What about calling this out in a note in the prerequisites? Then we could get rid of the comment about Calico in the Pod CIDR step, too. The note would look something like:

"The following examples run Calico as the Pod networking provider. If you run another networking provider, make sure to replace any default values as needed."

File: content/en/docs/setup/independent/high-availability.md [hunk 11]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
 
-## Acquire etcd certs
+## Setting up an HA etcd cluster using kubeadm
 
-Only follow this step if your etcd is hosted on dedicated nodes (**Option 1**). If you are hosting etcd on the masters (**Option 2**), you can skip this step since you've already generated the etcd certificates on the masters.
+Please see [setting up an HA etcd cluster using kubeadm](/docs/tasks/administer-cluster/setup-etcd-with-kubeadm).
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: lines 88-90 should be a numbered listitem instead:

`1. [Set up an HA etcd cluster using kubeadm](/docs/tasks/administer-cluster/setup-etcd-with-kubeadm).`

File: content/en/docs/setup/independent/high-availability.md [hunk 12]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-1. Run the following:
-     ```bash
-     mkdir -p /etc/kubernetes/pki/etcd
-     scp root@<etcd0-ip-address>:/etc/kubernetes/pki/etcd/ca.pem /etc/kubernetes/pki/etcd
-     scp root@<etcd0-ip-address>:/etc/kubernetes/pki/etcd/client.pem /etc/kubernetes/pki/etcd
-     scp root@<etcd0-ip-address>:/etc/kubernetes/pki/etcd/client-key.pem /etc/kubernetes/pki/etcd
-     ```
+1. `/etc/kubernetes/pki/etcd/ca.crt`
+1. `/etc/kubernetes/pki/apiserver-etcd-client.crt`
+1. `/etc/kubernetes/pki/apiserver-etcd-client.key`
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: ll 95-97 (each):
s/1./- (or your preferred character for a bulleted list)

File: content/en/docs/setup/independent/high-availability.md [hunk 13]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
 
-## Run `kubeadm init` on `master0` {#kubeadm-init-master0}
+Here is one way to do it, but there are many reasons why this might not work for
+you. For instance, you may be running any other linux distro.
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: (continue as block within the numbered list -- lmk if you need help figuring out the markup cuz it's different in Hugo/Blackfriday, although I think there might be OK models in some of the content you deleted). Also edit:

For example, you can run the following:

(note note that I suggest to follow)

File: content/en/docs/setup/independent/high-availability.md [hunk 14]
```diff
@@ -10,513 +10,217 @@ content_template: templates/task
...
-     - <load-balancer-ip>
-     apiServerExtraArgs:
-      apiserver-count: "3"
-     EOF
-     ```
+```
+scp /etc/kubernetes/pki/etcd/ca.crt ubuntu@ETCD_HOST_1:/home/ubuntu
+scp /etc/kubernetes/pki/apiserver-etcd-client.crt ubuntu@ETCD_HOST_1:/home/ubuntu
+scp /etc/kubernetes/pki/apiserver-etcd-client.key ubuntu@ETCD_HOST_2:/home/ubuntu
+```
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: Add note here:

"This is an example only. Your environment or Linux distribution might require different commands."

File: content/en/docs/setup/independent/high-availability.md (hunk 3 above)
Comment (chuckha) [reply content/en/docs/setup/independent/high-availability.md]: lol i was not even aware how many times i used certs, thanks for pointing this out.

**Comment by chuckha** [comment]
ptal @Bradamant3 @timothysc

**Comment by chuckha** [comment]
👀 on the ERRORs

**Comment by detiber** [comment]
/hold

I need to make a few tweaks to the stacked control plane config, will wrap up after family time :)

File: content/en/docs/setup/independent/high-availability.md [hunk 15]
```diff
@@ -1,536 +1,505 @@
...
-- errordeveloper
-- jbeda
+- timothysc
+- detiber
+- chuckha
 title: Creating HA clusters with kubeadm
 content_template: templates/task
 ---
 
+{{% capture steps %}}'
```
Comment (chuckha) [inline content/en/docs/setup/independent/high-availability.md]: I need help here. I don't want this but if I don't include a non-empty steps section then I end up with a big error page

Comment (chuckha) [reply content/en/docs/setup/independent/high-availability.md]: edit nvm! taken care of

**Comment by chuckha** [comment]
@Bradamant3 I integrated your changes as best I could but it has changed dramatically since you looked at it. please take a look when you've got time

File: content/en/docs/setup/independent/high-availability.md [hunk 16]
```diff
@@ -1,536 +1,502 @@
...
+This page explains two different ways to set up a highly available Kubernetes
+cluster using kubeadm:
 
-This document shows you how to perform setup tasks that kubeadm doesn't perform: provision hardware; configure multiple systems; and load balancing.
+- With stacked masters. This approach requires less infrastructure. etcd members
+and control plane nodes are co-located.
+- With an external etcd cluster. This approach requires more infrastructure. The
+control plane nodes and etcd members are separated.
+
+Your clusters must run Kubernetes version 1.11 or later.
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: @chuckha @Bradamant3 I think we need to reword this, but I'm failing on thinking of a better way to say that this doc will only work for v1.11 and later.

Also, should this be highlighted as a note to draw more attention to this?

File: content/en/docs/setup/independent/high-availability.md [hunk 17]
```diff
@@ -1,536 +1,502 @@
...
+address directly in a cloud environment). The load balancer must have the ability to
+communicate with all control plane nodes on the apiserver port and it must allow
+incoming traffic on its listening port.
+
+After this step, you have a resolvable DNS name for your load balancer.
+
+You should add the control plane nodes to the load balancer but be aware they
+will fail the health check until the kube-apiserver is running.
+
+## Stacked control plane nodes
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: @chuckha @Bradamant3 would it make sense (and is it even possible) to hide the two approaches here behind tabs, I think it would make the flow a bit better since there are common before/after steps for both.

File: content/en/docs/setup/independent/high-availability.md [hunk 18]
```diff
@@ -1,536 +1,502 @@
...
+  KUBECONFIG=/etc/kubernetes/admin.conf kubectl exec -n kube-system etcd-${CP0_HOSTNAME} -- etcdctl --ca-file /etc/kubernetes/pki/etcd/ca.crt --cert-file /etc/kubernetes/pki/etcd/peer.crt --key-file /etc/kubernetes/pki/etcd/peer.key --endpoints=https://${CP0_IP}:2379 member add ${CP1_HOSTNAME} https://${CP1_IP}:2380
+  kubeadm alpha phase etcd local --config kubeadm-config.yaml
+  ```
+
+1. Deploy the control plane components and mark the node as a master
+
+  ```sh
+  kubeadm alpha phase kubeconfig all --config kubeadm-config.yaml
+  kubeadm alpha phase controlplane all --config kubeadm-config.yaml
+  kubeadm alpha phase mark-master --config kubeadm-config.yaml
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: @Bradamant3 Technically, since we are requiring a load balancer for this and overriding the controlPlaneEndpoint these steps can be moved up above the etcd add step. I'm not sure if it makes sense to go ahead and consolidate these above or to leave them separate to make it more easily adaptable for the non-lb (or lb configured after install) approach...

**Comment by timothysc** [comment]
/approve 

I'll let @Bradamant3 + @detiber drive this one to done-done.

File: content/en/docs/setup/independent/high-availability.md (hunk 16 above)
Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: I haven't come up with anything better either, but I'll keep working on it.

**Comment by Bradamant3** [comment]
Preview build detail: https://deploy-preview-9066--kubernetes-io-vnext-staging.netlify.com/docs/setup/independent/high-availability/

Unfortunately, the tidier diff seems to have gone away with my commit. I'm still going to add some questions inline, but you might want to review the build instead.

I have not added the tabs. It's trivial to do, but I had to spend so much time getting lists to work that I didn't want to fiddle any more today.

**Comment by detiber** [comment]
@Bradamant3  thanks! I'll take a closer look at the changes in the morning.

File: content/en/docs/setup/independent/high-availability.md [hunk 19]
```diff
@@ -1,536 +1,517 @@
...
-- **Optional:** At least three machines that meet [kubeadm's minimum requirements](/docs/setup/independent/install-kubeadm/#before-you-begin)
-   if you intend to host etcd on dedicated nodes (see information below)
-- 1GB or more of RAM per machine (any less will leave little room for your apps)
-- Full network connectivity between all machines in the cluster (public or
-   private network is fine)
+{{< caution >}}
+**Caution**: This page does not address running your cluster on a cloud provider.
+In a cloud environment, neither approach documented here works with services of type 
+LoadBalancer, or with dynamic PersistentVolumes.
+{{< /caution >}}
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: I think I've captured the original meaning more clearly here. The only thing I'm wondering: are these "services" K8s Service objects? Would it be even clearer to say so? I'm assuming a pretty sophisticated/advanced audience, that understands the differences between these things and the load balancer/s that you need to set up for your control plane/s. But I had to think it through before I understood. I am not, however, the intended audience.

File: content/en/docs/setup/independent/high-availability.md [hunk 20]
```diff
@@ -65,48 +64,73 @@ needed.
...
+    (default value `:6443`).
 
-You should add the control plane nodes to the load balancer but be aware they
-will fail the health check until the kube-apiserver is running.
+    It is not recommended to use an IP address directly in a cloud environment.
 
-## Stacked control plane nodes
+    The load balancer must be able to communicate with all control plane node
+    on the apiserver port. It must also allow incoming traffic on its
+    listening port.
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: I trimmed some of the language about load balancer requirements, but it still seems unnecessarily repetitive. I might give it another go, but in the meantime if y'all could confirm that I've got the basic stuff here, that would be great.

File: content/en/docs/setup/independent/high-availability.md [hunk 21]
```diff
@@ -156,12 +181,11 @@ other control plane nodes. Specifically:
...
 
-The admin kubeconfig will also be needed for bootstrapping the additional
-control plane nodes:
+Copy the admin kubeconfig to the other control plane nodes:
 
 - `/etc/kubernetes/admin.conf`
 
-Here is an example, but your system may differ greatly. Replace
+In the following example, replace
 `CONTROL_PLANE_IP` with the IP addresses of the other control plane nodes.
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: I trimmed ruthlessly here. I don't think we need to keep explaining with every step that it's only an example.

File: content/en/docs/setup/independent/high-availability.md [hunk 22]
```diff
@@ -228,65 +250,59 @@ done
...
+      mv /home/${USER}/ca.crt /etc/kubernetes/pki/
+      mv /home/${USER}/ca.key /etc/kubernetes/pki/
+      mv /home/${USER}/sa.pub /etc/kubernetes/pki/
+      mv /home/${USER}/sa.key /etc/kubernetes/pki/
+      mv /home/${USER}/front-proxy-ca.crt /etc/kubernetes/pki/
+      mv /home/${USER}/front-proxy-ca.key /etc/kubernetes/pki/
+      mv /home/${USER}/etcd-ca.crt /etc/kubernetes/pki/etcd/ca.crt
+      mv /home/${USER}/etcd-ca.key /etc/kubernetes/pki/etcd/ca.key
+      mv /home/${USER}/admin.conf /etc/kubernetes/admin.conf
+      ```
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: The horrors of blocks inside listitems with Blackfriday, sigh.

File: content/en/docs/setup/independent/high-availability.md [hunk 23]
```diff
@@ -326,67 +342,70 @@ done
...
-
-  ```sh
-  kubeadm alpha phase certs all --config kubeadm-config.yaml
-  kubeadm alpha phase kubelet config write-to-disk --config kubeadm-config.yaml
-  kubeadm alpha phase kubelet write-env-file --config kubeadm-config.yaml
-  kubeadm alpha phase kubeconfig kubelet --config kubeadm-config.yaml
-  systemctl start kubelet
-  ```
+1. Move the copied certificates to the proper locations:
+
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: Urgh, note to self s/proper/correct

For reviewers: is there a reason we provide the copying information this way here, but in a loop script ^^? Consistency is a good thing in techdoc, if it's appropriate here ...

File: content/en/docs/setup/independent/high-availability.md [hunk 24]
```diff
@@ -396,17 +415,11 @@ done
          scp /etc/kubernetes/pki/apiserver-etcd-client.crt "${USER}"@$host:
          scp /etc/kubernetes/pki/apiserver-etcd-client.key "${USER}"@$host:
      done
```
Comment (Bradamant3) [inline content/en/docs/setup/independent/high-availability.md]: And we have a third way to copy certificates here. ???

File: content/en/docs/setup/independent/high-availability.md (hunk 19 above)
Comment (detiber) [reply content/en/docs/setup/independent/high-availability.md]: Services in this context are indeed Kubernetes Service Objects.

File: content/en/docs/setup/independent/high-availability.md (hunk 16 above)
Comment (chuckha) [reply content/en/docs/setup/independent/high-availability.md]: Maybe a "before you begin" section like this one? https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/

File: content/en/docs/setup/independent/high-availability.md [hunk 25]
```diff
@@ -129,23 +153,24 @@ will fail the health check until the kube-apiserver is running.
...
               - CP0_IP
         networking:
-            # This CIDR is a calico default. Substitute or remove for your CNI provider.
+            # This CIDR is a Calico default. Substitute or remove for your CNI provider.
             podSubnet: "192.168.0.0/16"
+        
 
-1. Replace the following variables in the template that was just created with
-   values for your specific situation:
+1.  Replace the following variables in the template with the appropriate
```
Comment (chuckha) [inline content/en/docs/setup/independent/high-availability.md]: oh my gracious this extra space after the number is a blackfriday thing isn't it?

Comment (neolit123) [reply content/en/docs/setup/independent/high-availability.md]: i think yes. blackfriday has(had?) bugs with the numbered lists, that are fixed upstream, but `k/website` hasn't updated yet, AFAIK.

**Comment by neolit123** [comment]
> @Bradamant3 Bradamant3 deleted a comment from...

hah :smile:

File: content/en/docs/setup/independent/high-availability.md (hunk 25 above)
Comment (mistyhacks) [reply content/en/docs/setup/independent/high-availability.md]: I prefer that extra space, actually. It means the words in the list item line up on a tab stop, and makes it easier to see how much to indent interior block-level content like code blocks (or in Black Friday, additional paragraphs in the list item).

**Comment by Bradamant3** [comment]
A couple of things to note about what I hope is the final version: 
- Spacing after numbers for numbered lists I realized belatedly is not consistent. Let's fix up later (I understand and agree with the argument for it).
- Tabs are a no-go for lengthy instructions. The layout is cramped and odd, and we lose what little we have left in the way of a decent generated TOC at the top of the file :-(.
- Maybe we can revisit breaking this into two files after the release. I wasn't part of the conversation that decided against this approach.

**Comment by Bradamant3** [comment]
We have an effective tech review in TimothySC's approve ^^, so

/lgtm

**Comment by mistyhacks** [comment]
I had to rebase 1.11 on master before the release so I rebased this PR to get it into good shape again.

File: content/en/docs/setup/independent/high-availability.md [hunk 26]
```diff
@@ -1,538 +1,517 @@
 ---
 reviewers:
-- mikedanese
 - luxas
-- errordeveloper
-- jbeda
-title: Creating HA clusters with kubeadm
+- timothysc
```
Comment (chuckha) [inline content/en/docs/setup/independent/high-availability.md]: this should be `sig-cluster-lifecycle` instead of names

File: content/en/docs/setup/independent/high-availability.md [hunk 27]
```diff
@@ -1,538 +1,517 @@
...
+    an apiserver is a TCP check on the port the kube-apiserver listens on 
+    (default value `:6443`).
+
+    It is not recommended to use an IP address directly in a cloud environment.
+
+    The load balancer must be able to communicate with all control plane nodes
+    on the apiserver port. It must also allow incoming traffic on its
+    listening port.
+
+1. Add the control plane nodes to the load balancer, but be aware they
```
Comment (chuckha) [inline content/en/docs/setup/independent/high-availability.md]: We should add a section to ensure our connections through the LB don't time out.

Something like 

1. Confirm the connection is working. A timeout confirms some settings need to be adjusted. A connection refused is ok because the API server is not running yet.
```
nc -v <load balancer ip> <port>
# example: nc -v 10.0.0.0 6443
```

Comment (rdodev) [reply content/en/docs/setup/independent/high-availability.md]: 💯 👍

File: content/en/docs/setup/independent/high-availability.md [hunk 28]
```diff
@@ -1,538 +1,525 @@
...
+
+1.  Replace the following variables in the template with the appropriate
+    values for your cluster:
+
+   * `LOAD_BALANCER_DNS`
+   * `LOAD_BALANCER_PORT`
+   * `CP0_HOSTNAME`
+   * `CP0_IP`
+
+1. Run `sudo kubeadm init --config kubeadm-config.yaml`
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: stray sudo left over, should be removed since we are telling users that commands should be run as root

Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: I'll remove in the commit I'm finishing now

File: content/en/docs/setup/independent/high-availability.md [hunk 29]
```diff
@@ -1,538 +1,524 @@
...
-            check_apiserver
-        }
-     }
-     ```
+      KUBECONFIG=/etc/kubernetes/admin.conf kubectl exec -n kube-system etcd-${CP0_HOSTNAME} -- etcdctl --ca-file /etc/kubernetes/pki/etcd/ca.crt --cert-file /etc/kubernetes/pki/etcd/peer.crt --key-file /etc/kubernetes/pki/etcd/peer.key --endpoints=https://${CP0_IP}:2379 member add ${CP1_HOSTNAME} https://${CP1_IP}:2380
+      kubeadm alpha phase etcd local --config kubeadm-config.yaml
+      ```
 
-    In the section `vrrp_instance VI_1`, change few lines depending on your setup:
+      {{< note >}}
```
[the remark this answers is not in the document]
Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: yup. Blackfriday/CSS FTW. I decided to live with it. If it drives you crazy, lmk and that'll make two of us :D

File: content/en/docs/setup/independent/high-availability.md [hunk 30]
```diff
@@ -1,538 +1,522 @@
...
-            check_apiserver
-        }
-     }
-     ```
+      KUBECONFIG=/etc/kubernetes/admin.conf kubectl exec -n kube-system etcd-${CP0_HOSTNAME} -- etcdctl --ca-file /etc/kubernetes/pki/etcd/ca.crt --cert-file /etc/kubernetes/pki/etcd/peer.crt --key-file /etc/kubernetes/pki/etcd/peer.key --endpoints=https://${CP0_IP}:2379 member add ${CP1_HOSTNAME} https://${CP1_IP}:2380
+      kubeadm alpha phase etcd local --config kubeadm-config.yaml
+      ```
 
-    In the section `vrrp_instance VI_1`, change few lines depending on your setup:
+      - This command causes the etcd cluster to become unavailable for a
```
Comment (detiber) [inline content/en/docs/setup/independent/high-availability.md]: I hate to say it, but I think the previous version was a bit better, this version doesn't really bring attention to the statement

Comment (Bradamant3) [reply content/en/docs/setup/independent/high-availability.md]: layout so ugly as to be distracting, though. I'll add it back in later.

**Comment by mistyhacks** [comment]
Thanks everyone for the hard work! 🎉 

/approve
